components/submit-expense/useExpenseForm.ts

```diff
@@ -696,7 +696,8 @@ function buildFormSchema(
       }, requiredMessage)
       .refine(
         accountSlug =>
-          accountSlug && (!options.supportedExpenseTypes || supportsBaseExpenseTypes(options.supportedExpenseTypes)),
+          accountSlug &&
+          (values.expenseTypeOption === ExpenseType.GRANT || supportsBaseExpenseTypes(options.supportedExpenseTypes)),
```

Comment:
Good fix, but we should also make sure that grants are supported:

```suggestion
          accountSlug &&
          (!options.supportedExpenseTypes ||
            supportsBaseExpenseTypes(options.supportedExpenseTypes) ||
            (values.expenseTypeOption === ExpenseType.GRANT &&
              options.supportedExpenseTypes.includes(ExpenseType.GRANT))),
```

Another issue here was apparently that, in this case, the error is properly generated but not surfaced in the interface.